src/main/java/com/phonebookservice/controller/ContactController.java

```diff
@@ -0,0 +1,55 @@
+package main.java.com.phonebookservice.controller;
+
+import main.java.com.phonebookservice.exception.PhoneBookControllerException;
+import main.java.com.phonebookservice.model.Contact;
+import main.java.com.phonebookservice.model.Model;
+import main.java.com.phonebookservice.server.DBIf;
+import main.java.com.phonebookservice.util.Messages;
+import main.java.com.phonebookservice.util.Util;
+
+public class ContactController extends AbstractController {
+
+    /**
+     * contact controller constructor.
+     *
+     * @param databse the database
+     */
+    public ContactController(final DBIf databse) {
+        super(databse);
+    }
+
+    @Override
+    public final void create(final Model model) {
```

Comment:
This method should take `Contact contact` not `Model model`. 

That's why Contact is inherited from Model. That's why inheritence is powerfull. 

You don't need to do in this method the following 

```
if (model == null || !(model instanceof Contact)) {
            throw new PhoneBookControllerException(
                    Messages.ERROR_OBJECT_INVALID);
        }
```